src/v/ssx/async_algorithm.h

```diff
@@ -106,7 +106,9 @@ template<std::random_access_iterator I, typename Fn>
 iter_size<I> for_each_limit(const I begin, const I end, ssize_t limit, Fn f) {
     auto chunk_size = std::min(limit, end - begin);
     I chunk_end = begin + chunk_size;
-    std::for_each(begin, chunk_end, std::move(f));
```

Comment:
I guess this doesn't honor it because warnings like that are suppressed when  the occur in system or stdlib headers?

Reply:
yes, I suppose so